### PR TITLE
Refactor EdgeConnect validation to v1alpha1

### PR DESCRIPTION
**Pull request by wepudt**
[K8S-10055](https://dt-rnd.atlassian.net/browse/K8S-10055)

## Description

Move the validation logic into EdgeConnect API version directory to allow for multi-version CRD validation.

## How can this be tested?

Deploy EdgeConnect with valid and invalid configurations to see if validation works.

### CODE REVIEW

File: cmd/webhook/builder.go [hunk 1]
```diff
@@ -179,7 +179,7 @@ func (builder CommandBuilder) buildRun() func(*cobra.Command, []string) error {
 			return err
 		}
 
-		err = edgeconnectvalidationhook.AddEdgeConnectValidationWebhookToManager(webhookManager)
+		err = edgeconnectv1alpha1validation.AddEdgeConnectValidationWebhookToManager(webhookManager)
```
Comment (andriisoldatenko) [inline cmd/webhook/builder.go:182]: should me same as for new dynakube:
```
package dynakube

import (
	ctrl "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
)

func SetupWebhookWithManager(mgr ctrl.Manager, validator admission.CustomValidator) error {
	return ctrl.NewWebhookManagedBy(mgr).
		For(&EdgeConnect{}).
		WithValidator(validator). // will create an endpoint at /validate-edgeconnect-com-v1beta1-edgeconnect
		Complete()
}

```

**Comment by andriisoldatenko** [comment]
don't forget to update https://github.com/dynatrace/dynatrace-operator/blob/fb0e89dad2cf49a7e1d537f0ccfaa0260225958d/config/helm/chart/default/templates/Common/webhook/validatingwebhookconfiguration.yaml#L64-L105

File: cmd/webhook/builder.go (hunk 1 above)
Comment (wepudt) [reply cmd/webhook/builder.go]: fixed in 6788c74cf9a36ee7b29223708f082dc163869b3e

**Comment by wepudt** [comment]
> don't forget to update https://github.com/dynatrace/dynatrace-operator/blob/fb0e89dad2cf49a7e1d537f0ccfaa0260225958d/config/helm/chart/default/templates/Common/webhook/validatingwebhookconfiguration.yaml#L64-L105

fixed in 6788c74cf9a36ee7b29223708f082dc163869b3e

**Review by aorcholski — APPROVED** [review]
LGTM can't create or update CR with invalid values

**Review by andriisoldatenko — APPROVED** [review]
practically LGTM, tested in my cluster.

**Pull request merged**